control/server.py

```diff
@@ -204,6 +214,38 @@ def _start_spdk(self):
         for trtype in spdk_transports.split():
             self._create_transport(trtype.lower())
 
+    def _stop_spdk(self):
+        """Stops SPDK process"""
+        assert self.spdk_process is not None # should be verified by the caller
+
+        return_code = self.spdk_process.returncode
+        rpc_socket = self.config.get("spdk", "rpc_socket")
+
+        # Terminate spdk process
+        if return_code is not None:
+            self.logger.error(f"SPDK({rpc_socket}) already terminated, exit code: {return_code}")
+        else:
+            self.logger.info("Terminating SPDK...")
+            self.spdk_process.terminate()
+        try:
+            timeout = self.config.getfloat("spdk", "timeout")
+            stdout, stderr = self.spdk_process.communicate(timeout=timeout)
+            for pair in [(stdout, "stdout"), (stderr, "stderr")]:
+                stream, name = pair
+                if stream is not None:
+                    for line in stream.split():
+                        self.logger.error(f"SPDK {name}: {line}")
+        except subprocess.TimeoutExpired as te:
+            self.logger.error(f"An timeout occurred while terminating spdk: {te}")
+            self.spdk_process.kill()
+
+        # Clean spdk rpc socket
+        if os.path.exists(rpc_socket):
+            try:
+                os.remove(rpc_socket)
+            except Exception as e:
+                self.logger.error(f"An error occurred while removing rpc socket {rpc_socket}: {e}")
```

Comment:
Are we intentionally muting any exception here?

Reply:
how?

Reply:
If you catch an exception but don't re-raise it again, the error will just consist of an error message, but the program flow will continue without changes (I personally feel that the `except` clause should always force to specify a `pass` or `raise` built-int to make it explicit whether the muting is intentional or not).

Reply:
Yeah, I could relate to this. Please consider, this code is potentially executed within exception handling code, such as in the case of  `TimeoutError`, always during application tear down. So it seems natural to handle exceptions, avoiding reraising in this case. Hope it makes sense.